### PR TITLE
[MTDSA-11875] List UK Savings Accounts

### PR DESCRIPTION
**Pull request by naree-hmrc**

### CODE REVIEW

File: app/v1/models/response/addUkSavingsAccount/AddUkSavingsAccountResponse.scala [hunk 1]
```diff
@@ -25,4 +25,6 @@ object AddUkSavingsAccountResponse {
   implicit val reads: Reads[AddUkSavingsAccountResponse] = (JsPath \ "incomeSourceId").read[String].map(AddUkSavingsAccountResponse(_))
 
   implicit val writes: Writes[AddUkSavingsAccountResponse] = Json.writes[AddUkSavingsAccountResponse]
+
+  // is the hateos link missing?
```
Comment (naree-hmrc) [inline app/v1/models/response/addUkSavingsAccount/AddUkSavingsAccountResponse.scala:29]: I think I found the answer for this. https://github.com/hmrc/individuals-income-received-api/pull/422

File: test/v1/models/response/listEmployment/EmploymentSpec.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ class EmploymentSpec extends UnitSpec {
 
     "return a valid Custom Employment model" when {
       "a valid custom employment json is supplied" in {
-        hmrcEmploymentJson.as[Employment] shouldBe hmrcEmploymentModel
+        customEmploymentJson.as[Employment] shouldBe customEmploymentModel
```
Comment (naree-hmrc) [inline test/v1/models/response/listEmployment/EmploymentSpec.scala:57]: Noticed that the assertion `hmrcEmploymentJson.as[Employment] shouldBe hmrcEmploymentModel` was duplicate and fixed it.

File: test/v1/models/response/listUkSavingsAccount/ListUkSavingsAccountSpec.scala [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      |   {
+      |      "incomeSourceId": "000000000000003",
+      |      "incomeSourceName": "Bank Account 3",
+      |      "identifier": "AA111111A",
+      |      "incomeSourceType": "interest-from-uk-banks"
+      |   }
+      |]
+    """.stripMargin)
+
+  val ukSavingsAccountsFromMDT = Json.parse(
```
Comment (sean-flynn-HMRC) [inline test/v1/models/response/listUkSavingsAccount/ListUkSavingsAccountSpec.scala:47]: What is MDT referring to? Is this meant to be "MTD" instead?

Comment (naree-hmrc) [reply test/v1/models/response/listUkSavingsAccount/ListUkSavingsAccountSpec.scala:47]: 👍

File: test/v1/models/response/listUkSavingsAccount/UkSavingsAccountSpec.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      "a valid uk savings account json from DES is supplied" in {
+        validUkSavingsAccountFromDESJson.as[UkSavingsAccount] shouldBe
+          UkSavingsAccount("SAVKB2UVwUTBQGJ", "Shares savings account")
+      }
+    }
+
+    "return a JsError" when {
+      "an invalid uk savings account json from DES is supplied" in {
+          invalidUkSavingsAccountFromDESJson.validate[UkSavingsAccount] shouldBe a[JsError]
+      }
```
Comment (sean-flynn-HMRC) [inline test/v1/models/response/listUkSavingsAccount/UkSavingsAccountSpec.scala:59]: Might it be worth also testing that a JsError is returned when an empty JSON is supplied?
